service/web/subscriberserver_test.go

```diff
@@ -684,3 +686,68 @@ func TestSubscriberServer_ListProjectSubscriptionsFailure(t *testing.T) {
 	assert.NotNil(t, err)
 	assert.Nil(t, res)
 }
+
+// TestSubscriberServer_GetSubscriptionWithExistingName: test the GetSubscription given subscription exist
+func TestSubscriberServer_GetSubscriptionWithExistingName(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockProjectCore := mocks4.NewMockICore(ctrl)
+	brokerStore := mocks.NewMockIBrokerStore(ctrl)
+	subscriptionCore := mocks2.NewMockICore(ctrl)
+	manager := mocks3.NewMockIManager(ctrl)
+	mockCredentialsCore := mocks5.NewMockICore(ctrl)
+	cache := cachemock.NewMockICache(ctrl)
+	server := newSubscriberServer(mockProjectCore, brokerStore, subscriptionCore, mockCredentialsCore, manager, cache)
+	ctx := context.Background()
+	req := &metrov1.GetSubscriptionRequest{
+		Name: "projects/project123/subscriptions/testsub",
+	}
+	sub := &subscription.Model{
+		Name:  "projects/project123/subscriptions/testsub",
+		Topic: "projects/project123/topics/test-topic",
+		PushConfig: &subscription.PushConfig{
+			PushEndpoint: "https://www.razorpay.com/api",
+			Attributes: map[string]string{
+				"test_key": "test_value",
+			},
+			Credentials: &credentials.Model{
+				Username: "test_user",
+				Password: "",
+			},
+		},
+		AckDeadlineSeconds: 10,
+		DeadLetterPolicy: &subscription.DeadLetterPolicy{
+			DeadLetterTopic:     "test_dlq",
+			MaxDeliveryAttempts: 1,
+		},
+		RetryPolicy: &subscription.RetryPolicy{
+			MinimumBackoff: 20,
+			MaximumBackoff: 300,
+		},
+	}
+	subscriptionCore.EXPECT().Get(gomock.Any(), req.GetName()).Times(1).Return(sub, nil)
+	res, err := server.GetSubscription(ctx, req)
+	pSub := subscription.ModelToSubscriptionProtoV1(sub)
+	assert.Nil(t, err)
+	assert.NotNil(t, res)
+	assert.True(t, reflect.DeepEqual(res, pSub))
+}
+
+// TestSubscriberServer_GetSubscriptionWithoutExistingName : test the GetSubscription given subscription does not exist
+func TestSubscriberServer_GetSubscriptionWithoutExistingName(t *testing.T) {
```

Comment:
The package_method format is used to measure coverage. Please do not add the test case identifiers here.

Reply:
corrected the format of test case